tests/python/gaia-ui-tests/gaiatest/tests/functional/ftu/test_ftu_skip_tour.py

```diff
@@ -80,11 +81,22 @@ def test_ftu_skip_tour(self):
         # Verify Geolocation section appears
         self.ftu.tap_next_to_geolocation_section()
 
-        # Disable geolocation
-        self.ftu.disable_geolocation()
-        self.wait_for_condition(
-            lambda m: not self.data_layer.get_setting('geolocation.enabled'),
-            message='Geolocation was not disabled by the FTU app')
+        # Verify the "geolocation" option matches expected initial state
+        # There's a very small lag on this after the dialog appears.
+        Wait(self.marionette).until(
+            lambda m: self.ftu.is_geolocation_enabled == geolocation_default,
+            message="Geolocation state should match initial internal state")
+
+        # Verify that internal state matches UI state
```

Comment:
Yeah, that was the conclusion in our daily standup as well. So I'll check this in and we can always change it when we do have integration tests on device.